templates/revoke.sh.j2

```diff
@@ -0,0 +1,43 @@
+#!/bin/sh
+
+
+CADIR="{{ certificates_ca_key_dir }}"
+
+cd ${CADIR}
+
+gen_crl () {
+  # regenerate the certificate revocation list
+  openssl ca -gencrl -config ca.conf -out ca-crl.pem
+}
+
+revoke () {
+  # revoke a client certificate
+  openssl ca -config ca.conf -revoke "${1}"
+}
+
+crl_out () {
+  # show certificate revocation list
+  openssl crl -in ca-crl.pem -noout -text
+}
+
+crl_verify () {
+  # verify that a certificate has been revoked
+  temp_pem="${mktemp}"
```

Comment:
Is this even tested? `${` uses a variable, `$(` executes a command

Reply:
yes its tested